### PR TITLE
snap: add contact: line in `snap info`

### PR DESCRIPTION
**Pull request by mvo5**
This results in the following output:
```
$ snap info godd
name:      godd
summary:   "godd is a nice dd replacement"
publisher: canonical
support: mailto:snappy-canonical-storeaccount@canonical.com
description: |
  godd is a nice improvement over dd as it has nice progress bars and
  knows about target devices
...
```
Happy to tweak the workding: `support: ` or `contact: ` ?

LP: #1624829

### CODE REVIEW

File: cmd/snap/cmd_info.go [hunk 1]
```diff
@@ -211,6 +211,9 @@ func (x *infoCmd) Execute([]string) error {
 		// TODO: have publisher; use publisher here,
 		// and additionally print developer if publisher != developer
 		fmt.Fprintf(w, "publisher:\t%s\n", both.Developer)
+		if both.SupportURL != "" {
+			fmt.Fprintf(w, "support: %s\n", both.SupportURL)
```
Comment (zyga) [inline cmd/snap/cmd_info.go]: Can you use `"support:\t%s\n"` like in the publisher line above?

File: snap/info.go [hunk 2]
```diff
@@ -124,6 +124,7 @@ type SideInfo struct {
 	EditedSummary     string   `yaml:"summary,omitempty" json:"summary,omitempty"`
 	EditedDescription string   `yaml:"description,omitempty" json:"description,omitempty"`
 	Private           bool     `yaml:"private,omitempty" json:"private,omitempty"`
+	SupportURL        string   `yaml:"support-url,omitempty" json:"support-url,omitempty"`
```
Comment (pedronis) [inline snap/info.go]: I have mixed feelings about this approach (get it from store and save in state), for one it means snaps on images will not get this unless we extend prepare-image and seed

File: cmd/snap/cmd_info.go (hunk 1 above)
Comment (mvo5) [inline cmd/snap/cmd_info.go]: ups, sorry! good catch.

File: snap/info.go (hunk 2 above)
Comment (mvo5) [inline snap/info.go]: Indeed, I'm not happy about this either. I can extend prepare-image/seeds (thanks for spotting this). But conceptually it feels like the state is not the best place. I'm struggling to find better alternatives though.

**Comment by mvo5** [comment]
We discussed this during the standup and the suggestion by Gustavo is to use `contact` and also put the `contact: ' into snap.yaml.

File: snap/info.go [hunk 3]
```diff
@@ -163,6 +162,8 @@ type Info struct {
 	Prices  map[string]float64 `yaml:"prices,omitempty" json:"prices,omitempty"`
 	MustBuy bool
 
+	Contact string
```
Comment (pedronis) [inline snap/info.go]: if it comes from snap.yaml it needs to go higher in this struct, there's a comment that says thing below come from the store only

**Comment by pedronis** [comment]
this will need store and snapcraft changes

**Comment by mvo5** [comment]
I set this to blocked right now, we need talk about if it goes into both snap.yaml or only into the store. In https://bugs.launchpad.net/snapcraft/+bug/1624829/comments/11 Mark suggests the store, in the standup Gustavo suggested the snap as well as the store.

**Comment by niemeyer** [comment]
People will tend to rebuild & push to the store without tuning that variable, and in general if one build/pushes some application, they should be the point of contact instead of the upstream developer. On that basis, let's please go with store-only for now.

We can always introduce snap.yaml support later, if that turns out to be mistaken.

**Comment by mvo5** [comment]
@niemeyer thank you! mailto: is now removed.

File: snap/info.go [hunk 4]
```diff
@@ -120,6 +120,7 @@ type SideInfo struct {
 	SnapID            string   `yaml:"snap-id" json:"snap-id"`
 	Revision          Revision `yaml:"revision" json:"revision"`
 	Channel           string   `yaml:"channel,omitempty" json:"channel,omitempty"`
+	Contact           string   `yaml:"contact,omitempty" json:"contact,omitempty"`
```
Comment (pedronis) [inline snap/info.go]: total personal nitpick, but I don't think the order is fully lexical here anyway, I would have put this before Private (like in seed)

**Pull request merged**